app/controllers/transactions_controller.rb

```diff
@@ -5,8 +5,13 @@ class TransactionsController < ApplicationController
 
   def new
     @q = current_organization.transactions.ransack(session[:filter])
-    @transaction = Transaction.new
-    @transfer = Transfer.new
+    if params[:invoice_id]
+      @transaction = Transaction.new(customer_name: current_organization.find_customer_name_by_id(params[:customer_id]),
+        amount: params[:amount], invoice_id: params[:invoice_id])
```

Comment:
@yunixon use `amount` and `customer` from `@invoice` object. Don't trust params from users.